collect_app/src/main/java/org/odk/collect/android/activities/GeoPolyActivity.java

```diff
@@ -341,11 +341,13 @@ private String formatPoints(List<MapPoint> points) {
         StringBuilder result = new StringBuilder();
         for (MapPoint point : points) {
             // TODO(ping): Remove excess precision when we're ready for the output to change.
-            result.append(String.format(Locale.US, "%s %s %s %s;",
+            result.append(String.format(Locale.US, "%s %s %s %s; ",
                     Double.toString(point.lat), Double.toString(point.lon),
                     Double.toString(point.alt), Float.toString((float) point.sd)));
         }
-        return result.toString().trim();
+
+        String answer = result.toString().trim();
+        return answer.isEmpty() ? answer : answer.substring(0, answer.length() - 1);
```

Comment:
@SaumiaSinghal 👋 heyo! What was the motivation behind this change? I'm a little worried changing the format of the answer could cause problems somewhere else?

Reply:
Hi Callum! yeah I was concerned about this too, so I mentioned this in regression risks as well. Previously, I was trying to match the `formatPoints` output to the `answerString`, so I found that `formatPoint` method does not add up spaces when adding another point, and the last point that is added in the string should not have semicolon at the end, in order to match with the `answerString`. I found this mentioned in #4451 as well.

Reply:
The general concept of trimming a semicolon is good. The PR should be updated to reflect that it also closes https://github.com/getodk/collect/issues/4451. I would strongly recommend using something like http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#removeEnd-java.lang.String-java.lang.String- to make it really explicit that a trailing ';' is removed.

Reply:
Some tests for the format of the underlying value format would also be great! Something like `pointsInGeotraces_areSeparatedBySemicolons`, `geotraces_doNotHaveTrailingSemicolons`, `geoshapes_haveMatchingFirstAndLastPoints` or something like that.

Reply:
Alright, I'll work on the tests.

Reply:
Hi @lognaturel! I was having some trouble as to where to add these tests, you mean for `GeoPolyActivity` test?

Reply:
Right, the current structure does not make this easy. For now I would pull `GeoPolyActivity#formatPoints` out to `GeoUtils`. There's already `formatLocationResultString` there and I think you could rename the new static method to `formatPointsResultString(List<MapPoint> points, boolean isShape)` or something like that. You can then add tests on the static method in `GeoUtilsTest`. It's not great to introduce more static methods but I think it's better than leaving this functionality untested.